cmd/server/main.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 Zhizhesihai (Beijing) Technology Limited.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"net/http"
+	"os"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/juju/loggo"
+	"github.com/urfave/cli/v2"
+	v1 "github.com/zhihu/cmdb/pkg/api/v1"
+	"github.com/zhihu/cmdb/pkg/signals"
+	"github.com/zhihu/cmdb/pkg/storage/tidb"
+	"github.com/zhihu/cmdb/pkg/tools/grpcserver"
+	"github.com/zhihu/cmdb/pkg/tools/httpserver"
+	"github.com/zhihu/cmdb/pkg/tools/logger"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+)
+
+var log = loggo.GetLogger("server")
+
+const (
+	FlagLogConf  = "log_conf"
+	FlagGrpcAddr = "grpc"
+	FlagHTTPAddr = "http"
+	FlagDSN      = "dsn"
+)
+
+func main() {
+	ctx := signals.SignalHandler(context.Background())
+	// Register gRPC server endpoint
+	// Note: Make sure the gRPC server is running properly and accessible
+
+	app := cli.NewApp()
+
+	app.Flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:    FlagLogConf,
+			Usage:   "config logger level,see http://github.com/juju/loggo",
+			EnvVars: []string{"LOG_CONF"},
+			Value:   "<root>=INFO",
+		},
+		&cli.StringFlag{
```

Comment:
we can use github.com/soheilhy/cmux to serve request for both gRPC and HTTP on the same port.